src/coreclr/jit/compiler.cpp

```diff
@@ -5106,9 +5104,27 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     compJitTelemetry.NotifyEndOfCompilation();
 #endif
 
-#if defined(DEBUG)
-    ++Compiler::jitTotalMethodCompiled;
-#endif // defined(DEBUG)
+    unsigned methodsCompiled = (unsigned)InterlockedIncrement((LONG*)&Compiler::jitTotalMethodCompiled);
+
+    if (JitConfig.DumpJittedMethods() && !compIsForInlining())
+    {
+#ifdef DEBUG
+        const int BUFSIZE            = 20;
+        char      osrBuffer[BUFSIZE] = {0};
+        if (opts.IsOSR())
+        {
+            // Tiering name already includes "OSR", we just want the IL offset
+            sprintf_s(osrBuffer, BUFSIZE, " @0x%x", info.compILEntry);
+        }
+
+        printf("%4d: JIT compiled %s [%s%s, IL size=%u, code size=%u, hash=0x%08x %s]\n", methodsCompiled,
+               info.compFullName, compGetTieringName(), osrBuffer, info.compILCodeSize, *methodCodeSize,
+               info.compMethodHash(), compGetStressMessage());
```

Comment:
`compGetStressMessage` includes a leading space if necessary. Also, maybe move the stress message right after the Tier/OSR string, since it indicates another "flavor" of compilation:

```suggestion
        printf("%4d: JIT compiled %s [%s%s%s, IL size=%u, code size=%u, hash=0x%08x]\n", methodsCompiled,
               info.compFullName, compGetTieringName(), osrBuffer, compGetStressMessage(), info.compILCodeSize, *methodCodeSize,
               info.compMethodHash());
```